UnrealPlugin/Source/PlayfabGSDK/Private/FGSDKInternal.cpp

```diff
@@ -0,0 +1,374 @@
+			{
+				if (ConfigSettings.Contains(SessionConfigJsonValue.Key))
+				{
+					ConfigSettings[SessionConfigJsonValue.Key] = ValueString;
+				}
+				else
+				{
+					ConfigSettings.Add(SessionConfigJsonValue.Key, ValueString);
+				}
+			}
+		}		
+
+		// Update initial players only if this is the first time populating it.
+		if (InitialPlayers.Num() == 0 && SessionConfigJson->HasField(TEXT("initialPlayers")))
+		{
+			TArray<TSharedPtr<FJsonValue>> PlayersJson = SessionConfigJson->GetArrayField(TEXT("initialPlayers"));
+
+			for (auto PlayerJson: PlayersJson)
+			{
+				InitialPlayers.Add(PlayerJson->AsString());
+			}
+		}
+
+		if (SessionConfigJson->HasField(TEXT("metadata")))
+		{
+			TSharedPtr<FJsonObject> MetaDatasJson = SessionConfigJson->GetObjectField(TEXT("metadata"));
+
+			for (const auto& MetaDataJson: MetaDatasJson->Values)
+			{
+				FString ValueString;
+				if (MetaDataJson.Value->TryGetString(ValueString))
+				{
+					if (ConfigSettings.Contains(MetaDataJson.Key))
+					{
+						ConfigSettings[MetaDataJson.Key] = ValueString;
+					}
+					else
+					{
+						ConfigSettings.Add(MetaDataJson.Key, ValueString);
+					}
+				}
+			}
+		}
+	}
+
+	if (HeartbeatResponseJson->HasField("nextScheduledMaintenanceUtc"))
+	{
+		FDateTime NextMaintenance = ParseDate(HeartbeatResponseJson->GetStringField(TEXT("nextScheduledMaintenanceUtc")));
+		FTimespan Diff = NextMaintenance - CachedScheduledMaintenance;
+
+		if (OnMaintenance.IsBound() && (!Diff.IsZero() || CachedScheduledMaintenance.GetTicks()))
+		{
+			OnMaintenance.Execute(NextMaintenance);
+			CachedScheduledMaintenance = NextMaintenance;
+		}
+	}
+
+	if (HeartbeatResponseJson->HasField(TEXT("operation")))
+	{
+		EOperation NextOperation = OperationMap[HeartbeatResponseJson->GetStringField(TEXT("operation"))];
+
+		switch (NextOperation)
+		{
+		case EOperation::Continue:
+			{
+				break;
+			}
+		case EOperation::Active:
+			{
+				if (HeartbeatRequest.CurrentGameState != EGameState::Active)
+				{
+					SetState(EGameState::Active);
+					TransitionToActiveEvent->Trigger();
+				}
+				break;
+			}
+		case EOperation::Terminate:
+			{
+				if (HeartbeatRequest.CurrentGameState != EGameState::Terminating)
+				{
+					SetState(EGameState::Terminating);
+					TransitionToActiveEvent->Trigger();
+					AsyncTask(ENamedThreads::AnyThread, [this]()
+					{
+						if (this->OnShutdown.IsBound())
+						{
+							this->OnShutdown.Execute();
+						}
+
+						this->KeepHeartbeatRunning = false;
+					});
+				}
+				break;
+			}
+		default:
+			{
+				UE_LOG(LogPlayfabGSDK, Error, TEXT("Unhandled operation received: %s"), OperationNames[static_cast<int32>(NextOperation)]);
+				break;
+			}
+		}
+	}
+
+	if (HeartbeatResponseJson->HasField(TEXT("nextHeartbeatIntervalMs")))
+	{
+		NextHeartbeatIntervalMs = HeartbeatResponseJson->GetNumberField(TEXT("nextHeartbeatIntervalMs"));
+
+		
+		// Clamp to the minimum permitted interval.
```

Comment:
https://docs.unrealengine.com/4.26/en-US/API/Runtime/Core/Math/FMath/Clamp/

Reply:
changed to FMath::Max